scipy/ndimage/tests/test_fourier.py

```diff
@@ -1,137 +1,181 @@
+    def test_fourier_uniform_real01(self, shape, dtype, dec, xp):
+        fft = getattr(xp, 'fft')
+
+        a = np.zeros(shape, dtype=dtype)
         a[0, 0] = 1.0
-        a = fft.rfft(a, shape[0], 0)
-        a = fft.fft(a, shape[1], 1)
+        a = xp.asarray(a)
+
+        a = fft.rfft(a, n=shape[0], axis=0)
+        a = fft.fft(a, n=shape[1], axis=1)
         a = ndimage.fourier_uniform(a, [5.0, 2.5], shape[0], 0)
-        a = fft.ifft(a, shape[1], 1)
-        a = fft.irfft(a, shape[0], 0)
-        assert_almost_equal(ndimage.sum(a), 1.0, decimal=dec)
+        a = fft.ifft(a, n=shape[1], axis=1)
+        a = fft.irfft(a, n=shape[0], axis=0)
+        assert_almost_equal(ndimage.sum(a), xp.asarray(1.0), decimal=dec)
 
     @pytest.mark.parametrize('shape', [(32, 16), (31, 15)])
-    @pytest.mark.parametrize('dtype, dec', [(np.complex64, 6), (np.complex128, 14)])
-    def test_fourier_uniform_complex01(self, shape, dtype, dec):
-        a = np.zeros(shape, dtype)
+    @pytest.mark.parametrize('dtype, dec', [("complex64", 6), ("complex128", 14)])
+    def test_fourier_uniform_complex01(self, shape, dtype, dec, xp):
+        fft = getattr(xp, 'fft')
+
+        a = np.zeros(shape, dtype=dtype)
         a[0, 0] = 1.0
-        a = fft.fft(a, shape[0], 0)
-        a = fft.fft(a, shape[1], 1)
+        a = xp.asarray(a)
+
+        a = fft.fft(a, n=shape[0], axis=0)
+        a = fft.fft(a, n=shape[1], axis=1)
         a = ndimage.fourier_uniform(a, [5.0, 2.5], -1, 0)
-        a = fft.ifft(a, shape[1], 1)
-        a = fft.ifft(a, shape[0], 0)
-        assert_almost_equal(ndimage.sum(a.real), 1.0, decimal=dec)
+        a = fft.ifft(a, n=shape[1], axis=1)
+        a = fft.ifft(a, n=shape[0], axis=0)
+        assert_almost_equal(ndimage.sum(xp.real(a)), xp.asarray(1.0), decimal=dec)
 
     @pytest.mark.parametrize('shape', [(32, 16), (31, 15)])
-    @pytest.mark.parametrize('dtype, dec', [(np.float32, 4), (np.float64, 11)])
-    def test_fourier_shift_real01(self, shape, dtype, dec):
-        expected = np.arange(shape[0] * shape[1], dtype=dtype)
-        expected.shape = shape
-        a = fft.rfft(expected, shape[0], 0)
-        a = fft.fft(a, shape[1], 1)
+    @pytest.mark.parametrize('dtype, dec', [("float32", 4), ("float64", 11)])
+    def test_fourier_shift_real01(self, shape, dtype, dec, xp):
+        fft = getattr(xp, 'fft')
+
+        expected = np.arange(shape[0] * shape[1], dtype=dtype).reshape(shape)
+        expected = xp.asarray(expected)
+
+        a = fft.rfft(expected, n=shape[0], axis=0)
+        a = fft.fft(a, n=shape[1], axis=1)
         a = ndimage.fourier_shift(a, [1, 1], shape[0], 0)
-        a = fft.ifft(a, shape[1], 1)
-        a = fft.irfft(a, shape[0], 0)
+        a = fft.ifft(a, n=shape[1], axis=1)
+        a = fft.irfft(a, n=shape[0], axis=0)
         assert_array_almost_equal(a[1:, 1:], expected[:-1, :-1], decimal=dec)
-        assert_array_almost_equal(a.imag, np.zeros(shape), decimal=dec)
```

Comment:
why is this removed?

Reply:
https://github.com/scipy/scipy/pull/21150#discussion_r1675793114